File: Problem014-Longest Common Prefix/Solution.py
"""
 @SEU(master);@ZZULI(bachelor)
 author:CuiXuan
 email:873059043@qq.com
 If you find any errors, please let me know.
 If you have any better solution, please let me know.

 Solution Using time 66ms
"""

class Solution:
    # @param {string[]} strs
    # @return {string}
    def longestCommonPrefix(self, strs):
        if len(strs)==0:return ""
        str=strs[0]
        Min=len(str)
        for i in range(1,len(strs)):
            j=0;p=strs[i]
            while j<Min and j<len(p) and p[j]==str[j]:j+=1
            Min = Min if Min<j else j
        return str[:Min]

File: Problem014-Longest Common Prefix/Solution.c
/*
 * @SEU(master);@ZZULI(bachelor)
 * author:CuiXuan
 * email:873059043@qq.com
 * If you find any errors, please let me know.
 * If you have any better solution, please let me know.
 *
 * Solution Using time 6ms
 */
char* longestCommonPrefix(char** strs, int strsSize) {
    char* str=strs[0];
    int i,j;
    if(strsSize==0)return "";
    for(i=1;i<strsSize;i++){
        j=0;
        while(str[j] && strs[i][j] && str[j]==strs[i][j])j++;
        str[j]=0;
    }
    return str;
}
